### DISCUSSION TITLE
Audit log of private repository downloads

### DISCUSSION THREAD

**Discussion by sumit-creates** [community/community#23153] [New to GitHub]
<p>Hello,</p>
<p>Is it possible to get detailed audit log of downloads (e.g. performed using pull/fetch/clone commands) of private repositories on the GitHub?</p>
<p>Specifically, information identifying user, command used, date &amp; time, and destination URL/IP address will be very useful in identification of any suspicious activity.</p>
<p>SM</p>

**Comment by byrneh** [comment, marked as answer]
<p><a class="mention" href="/sumit-creates">@sumit-creates</a> unfortunately the Audit event log API does not include events of Git actions. It is a feature requested by many people, and still nothing on the public roadmap for this.</p>
<p>There are Webhook events for git actions. Not as user friendly, you would need to configure and setup webhook’s and a listener to pick up the http payloads sent to your Webhook listener. You will also need to consider how you wish to handle lost/undelivered messages</p>

**Reply by smarj** [reply]
This response is outdated, please see [below](https://github.com/orgs/community/discussions/23153#discussioncomment-6177711).

**Comment by sumit-creates** [comment]
<p>Hi Hugh,</p>
<p>Thanks for your prompt and helpful response! Appreciate your suggestion about the Webhook events plus git actions based alternative approach.</p>
<p>Yes, it definitely seems complicated to automate this apparently simple yet important task.  It will be great if GitHub makes this more user and automation friendly given that sharing of sensitive data in private repositories mandates active monitoring.</p>
<p>Thanks and best regards,<br>
Sumit</p>

**Comment by byrneh** [comment]
<p>See these roadmap updates</p><aside class="quote quote-modified" data-post="22" data-topic="13783">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/what-is-the-best-way-to-acquire-auditlog-in-organization/13783/22">What is the best way to acquire AuditLog in organization?</a> <a class="badge-wrapper  bullet" href="/c/github-ecosystem/37"><span class="badge-category-bg" style="background-color: #28a745;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and support using GitHub’s REST, and GraphQL APIs, building Applications and OAuth Apps, the GitHub Marketplace, and third party integrations.">GitHub Ecosystem</span></a>
  </div>
  <blockquote>
    Here are two roadmap features relevant to this topic 
<a href="https://github.com/github/roadmap/issues/109" rel="noopener nofollow ugc">Git action events (e.g. clone and push) appear in the audit logs</a> (beta Q4 2020) 
GitHub adding Git CLI activities to the audit log on Github Enterprise Cloud 
<a href="https://github.com/github/roadmap/issues/125" rel="noopener nofollow ugc">Webhook Delivery API (beta)</a> (beta Q1 2021) 
Today, webhooks may fail to be delivered for a variety of reasons: severed connections, downtime from GitHub, downtime from the integrator, etc. Right now, we offer the ability to view failed webhook deliveries in the UI and retry them, but we…
  </blockquote>
</aside>

**Comment by sumit-creates** [comment]
<p>Thanks Hugh for sharing this very important and relevant information.</p>
<p>Looking forward to try these new additions. It seems from the description that our primary objective of monitoring sensitive information usage will be achieved to a significant extent with the help of mentioned features.</p>
<p>Thanks and best regards,<br>
Sumit</p>

**Comment by byrneh** [comment]
<p><a href="https://github.community/t/new-feature-audit-log-rest-api-check-it-out/161512">New Feature Audit Log REST API check it out</a></p>

**Comment by smarj** [comment]
This has long since been resolved: [Audit Log Git events and REST API are generally available | GitHub Changelog](https://github.blog/changelog/2021-04-12-audit-log-git-events-and-rest-api-are-generally-available/).

**Comment by uday-infosec** [comment]
Hey @smarj / @byrneh, has this feature been made available for on-prem enterprise git and git enterprise SaaS ? The above link does not provide clarity on git clones, pulls, fetch and tarball/zip downloads. Could you please clarify ?

**Comment by MosJef** [comment]
> Hey @smarj / @byrneh, has this feature been made available for on-prem enterprise git and git enterprise SaaS ? The above link does not provide clarity on git clones, pulls, fetch and tarball/zip downloads. Could you please clarify ?

If you follow the link in the article to the docs, you can select your enterprise version (including Enterprise Server) which will show the specifics for the version you are running. [Enterprise Server 3.9 - Using the audit log API](https://docs.github.com/en/enterprise-server@3.9/organizations/keeping-your-organization-secure/managing-security-settings-for-your-organization/reviewing-the-audit-log-for-your-organization#using-the-audit-log-api)

The same can be said with [Enterprise Server 3.9 - Audit log events for your organization](https://docs.github.com/en/enterprise-server@3.9/organizations/keeping-your-organization-secure/managing-security-settings-for-your-organization/audit-log-events-for-your-organization). Which shows `git.fetch` `git.clone` and `git.push` as available events.
